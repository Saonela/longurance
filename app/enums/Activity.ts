export enum Activity {
    SWIMMING = 'SWIMMING',
    CYCLING = 'CYCLING',
    RUNNING = 'RUNNING'
}

export const ActivityOptions: {label: string; value: Activity}[] = [
    {label: 'Running', value: Activity.RUNNING},
    {label: 'Swimming', value: Activity.SWIMMING},
    {label: 'Cycling', value: Activity.CYCLING}
];
