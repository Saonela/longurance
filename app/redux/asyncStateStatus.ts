export enum ASYNC_STATE_STATUS {
    IDLE,
    LOADING,
    SUCCEEDED,
    FAILED
}
