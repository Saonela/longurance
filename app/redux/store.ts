import {useDispatch} from 'react-redux';
import {configureStore} from '@reduxjs/toolkit';
import entriesReducer from './slices/entriesSlice';
import entriesFilterReducer from './slices/entriesFilterSlice';
import trophiesReducer from './slices/trophiesSlice';
import statisticsOptionsReducer from './slices/statisticsOptionsSlice';

const store = configureStore({
    reducer: {
        entriesFilter: entriesFilterReducer,
        entries: entriesReducer,
        trophies: trophiesReducer,
        statisticsOptions: statisticsOptionsReducer
    }
});

export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = () => useDispatch<AppDispatch>(); // Export a hook that can be reused to resolve types

export default store;
