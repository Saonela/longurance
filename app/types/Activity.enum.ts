export enum Activity {
    SWIMMING = 'SWIMMING',
    CYCLING = 'CYCLING',
    RUNNING = 'RUNNING'
}
