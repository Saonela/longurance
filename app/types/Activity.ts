export enum Activity {
    SWIMMING = 'SWIMMING',
    CYCLING = 'CYCLING',
    RUNNING = 'RUNNING'
}

export const ActivityOptions: {label: string, value: string}[] = [
    {label: 'Running', value: Activity.RUNNING},
    {label: 'Swimming', value: Activity.SWIMMING},
    {label: 'Cycling', value: Activity.CYCLING}
];
