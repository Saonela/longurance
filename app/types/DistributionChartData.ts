export type DistributionChartData = {
    label: string;
    value: number;
}[];
