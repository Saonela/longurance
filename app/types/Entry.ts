import {Activity} from './Activity';

export interface Entry {
    id: string;
    createdAt: string;
    activity: Activity;
    distance: number;
    duration: number;
    date: string;
    effort: number;
    note: string;
    title: string;
}
