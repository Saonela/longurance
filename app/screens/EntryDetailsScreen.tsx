import React, {useLayoutEffect} from 'react';
import {Alert, ScrollView, View} from 'react-native';
import HeaderButton from '../components/header/HeaderButton';
import appStyles from '../styles';
import EntryDetails from '../components/entry/EntryDetails';
import {deleteEntry, getEntry, useEntriesStore} from '../state/entries';
import {Entry} from '../types/Entry';
import {
    getEntryTrophies,
    updateCompletedTrophies,
    useTrophiesStore
} from '../state/trophies';
import theme from '../theme';
import TrophyCardLite from '../components/trophy/TrophyCardLite';
import {TrophyType} from '../enums/TrophyType';

function EntryDetailsScreen({route, navigation}) {
    const entry = useEntriesStore((state) =>
        getEntry(state, route.params.id)
    ) as Entry;

    const trophies = useTrophiesStore(
        getEntryTrophies(entry?.id, TrophyType.INDIVIDUAL)
    );

    useLayoutEffect(() => {
        const confirmDelete = () => {
            Alert.alert(
                'Delete entry',
                'Entry will be removed from history',
                [
                    {
                        text: 'Cancel',
                        style: 'cancel'
                    },
                    {
                        text: 'OK',
                        onPress: async () => {
                            deleteEntry(entry.id);
                            updateCompletedTrophies();
                            navigation.goBack();
                        }
                    }
                ],
                {cancelable: false}
            );
        };

        navigation.setOptions({
            headerTitleStyle: {
                maxWidth: 200
            },
            headerRight: () => (
                <View style={{display: 'flex', flexDirection: 'row'}}>
                    <HeaderButton
                        iconName="edit"
                        onPress={() =>
                            navigation.navigate('entry-form', {id: entry.id})
                        }
                    />
                    <HeaderButton iconName="x" onPress={confirmDelete} />
                </View>
            )
        });
    }, [navigation, entry?.id]);

    return (
        <View style={appStyles.screenContainer}>
            <ScrollView
                contentContainerStyle={{paddingBottom: theme.SPACING.M}}
            >
                {entry && <EntryDetails entry={entry} />}
                {trophies.map((trophy) => (
                    <TrophyCardLite key={trophy.id} trophy={trophy} />
                ))}
            </ScrollView>
        </View>
    );
}

export default EntryDetailsScreen;
