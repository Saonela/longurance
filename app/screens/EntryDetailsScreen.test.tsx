import {configureStore} from '@reduxjs/toolkit';
import {ASYNC_STATE_STATUS} from '../redux/asyncStateStatus';
import React from 'react';
import {Provider} from "react-redux";
import {render} from '@testing-library/react-native';
import entriesReducer from '../redux/slices/entriesSlice';
import entriesFilterReducer from '../redux/slices/entriesFilterSlice';
import {Activity} from '../types/Activity';
import {Entry} from '../types/Entry';
import EntryDetailsScreen from './EntryDetailsScreen';
import renderer from 'react-test-renderer';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');


const entries: Partial<Entry>[] = [
    {
        id: '1',
        activity: Activity.RUNNING,
        distance: 15,
        duration: 4000,
        date: '2021-01-07T09:10:02.207Z',
        energy: 2,
        title: 'YES',
        note: 'Was really enjoying. Got into flow state.',
    }
];

describe('EntryDetailsScreen', () => {
    it('should show entry details', () => {
        const initialState: any = {
            entries: {
                status: ASYNC_STATE_STATUS.SUCCEEDED,
                error: null,
                data: entries
            },
            entriesFilter: null
        };
        const store = configureStore({
            reducer: {
                entries: entriesReducer,
                entriesFilter: entriesFilterReducer
            },
            preloadedState: initialState
        });

        const navigation = {
            setOptions: () => {}
        }

        const component = (
            <Provider store={store}>
                <EntryDetailsScreen navigation={navigation} route={{params: {id: '1'}}}/>
            </Provider>
        );

        const {getByText} = render(component);

        getByText('Was really enjoying. Got into flow state.');
    });

    it('should match snapshot', () => {
        const initialState: any = {
            entries: {
                status: ASYNC_STATE_STATUS.SUCCEEDED,
                error: null,
                data: entries
            },
            entriesFilter: null
        };
        const store = configureStore({
            reducer: {
                entries: entriesReducer,
                entriesFilter: entriesFilterReducer
            },
            preloadedState: initialState
        });

        const navigation = {
            setOptions: () => {}
        }

        const component = (
            <Provider store={store}>
                <EntryDetailsScreen navigation={navigation} route={{params: {id: '1'}}}/>
            </Provider>
        );

        const tree = renderer
            .create(component)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});
