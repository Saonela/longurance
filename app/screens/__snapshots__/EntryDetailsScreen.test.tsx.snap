// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryDetailsScreen should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#121212",
        "flex": 1,
      }
    }
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#2a2a2a",
              "borderRadius": 16,
              "height": 120,
              "margin": 8,
              "overflow": "hidden",
              "padding": 16,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#f4f4f6",
                    "fontFamily": "Lato",
                    "fontSize": 13,
                    "marginRight": 8,
                  }
                }
              >
                2021
              </Text>
              <View
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#f4f4f6",
                      "fontFamily": "Lato",
                      "fontSize": 22,
                      "marginRight": 8,
                    }
                  }
                >
                  Jan 07
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#f4f4f6",
                      "fontFamily": "Lato",
                      "fontSize": 15,
                      "marginRight": 8,
                      "marginTop": 6,
                    }
                  }
                >
                  16:10
                </Text>
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "bottom": 16,
                "color": "#f4f4f6",
                "fontSize": 24,
                "left": 16,
                "position": "absolute",
              }
            }
          >
            RUN
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#4b4b4e",
                    "fontSize": 200,
                  },
                  Object {
                    "color": "#4b4b4e",
                    "position": "absolute",
                    "right": -30,
                    "top": -90,
                  },
                  Object {
                    "fontFamily": "FontAwesome5Free-Solid",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontWeight": "700",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#2a2a2a",
                  "borderRadius": 16,
                  "flexGrow": 1,
                  "margin": 8,
                  "padding": 16,
                },
                Object {
                  "flexBasis": 1,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "position": "absolute",
                  "right": 16,
                  "top": "50%",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#f4f4f6",
                      "fontSize": 32,
                    },
                    undefined,
                    Object {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                󰣰
              </Text>
            </View>
            <Text
              style={
                Object {
                  "color": "#f4f4f6",
                  "fontFamily": "Lato",
                  "fontSize": 15,
                  "marginBottom": 4,
                }
              }
            >
              Distance
            </Text>
            <Text
              style={
                Object {
                  "color": "#f4f4f6",
                  "fontFamily": "Lato",
                  "fontSize": 15,
                  "maxWidth": "90%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#f4f4f6",
                      "fontFamily": "Lato",
                      "fontSize": 15,
                    },
                    Object {},
                  ]
                }
              >
                15 KM
              </Text>
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#2a2a2a",
                  "borderRadius": 16,
                  "flexGrow": 1,
                  "margin": 8,
                  "padding": 16,
                },
                Object {
                  "flexBasis": 1,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "position": "absolute",
                  "right": 16,
                  "top": "50%",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#f4f4f6",
                      "fontSize": 32,
                    },
                    undefined,
                    Object {
                      "fontFamily": "material",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <Text
              style={
                Object {
                  "color": "#f4f4f6",
                  "fontFamily": "Lato",
                  "fontSize": 15,
                  "marginBottom": 4,
                }
              }
            >
              Duration
            </Text>
            <Text
              style={
                Object {
                  "color": "#f4f4f6",
                  "fontFamily": "Lato",
                  "fontSize": 15,
                  "maxWidth": "90%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#f4f4f6",
                      "fontFamily": "Lato",
                      "fontSize": 15,
                    },
                    Object {},
                  ]
                }
              >
                01:06:40
              </Text>
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#2a2a2a",
                  "borderRadius": 16,
                  "flexGrow": 1,
                  "margin": 8,
                  "padding": 16,
                },
                Object {
                  "flexBasis": 1,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "position": "absolute",
                  "right": 16,
                  "top": "50%",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#f4f4f6",
                      "fontSize": 32,
                    },
                    undefined,
                    Object {
                      "fontFamily": "material",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <Text
              style={
                Object {
                  "color": "#f4f4f6",
                  "fontFamily": "Lato",
                  "fontSize": 15,
                  "marginBottom": 4,
                }
              }
            >
              Pace
            </Text>
            <Text
              style={
                Object {
                  "color": "#f4f4f6",
                  "fontFamily": "Lato",
                  "fontSize": 15,
                  "maxWidth": "90%",
                }
              }
            >
              <Text>
                KM / 
              </Text>
              <View />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#f4f4f6",
                      "fontFamily": "Lato",
                      "fontSize": 15,
                    },
                    Object {},
                  ]
                }
              >
                00:04:26.666666666666686
              </Text>
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#2a2a2a",
                  "borderRadius": 16,
                  "flexGrow": 1,
                  "margin": 8,
                  "padding": 16,
                },
                Object {
                  "flexBasis": 1,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "position": "absolute",
                  "right": 16,
                  "top": "50%",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#f4f4f6",
                      "fontSize": 32,
                    },
                    undefined,
                    Object {
                      "fontFamily": "simple-line-icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <Text
              style={
                Object {
                  "color": "#f4f4f6",
                  "fontFamily": "Lato",
                  "fontSize": 15,
                  "marginBottom": 4,
                }
              }
            >
              Trophy
            </Text>
            <Text
              style={
                Object {
                  "color": "#f4f4f6",
                  "fontFamily": "Lato",
                  "fontSize": 15,
                  "maxWidth": "90%",
                }
              }
            >
              -
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#2a2a2a",
              "borderRadius": 16,
              "flexGrow": 1,
              "margin": 8,
              "padding": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#f4f4f6",
                "fontFamily": "Lato",
                "fontSize": 15,
                "marginBottom": 4,
              }
            }
          >
            Effort
          </Text>
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4283023632,
                  4292664884,
                  4294914867,
                ],
                "endPoint": Array [
                  1,
                  0,
                ],
                "locations": undefined,
                "startPoint": Array [
                  0,
                  0,
                ],
              }
            }
            style={
              Object {
                "backgroundColor": "#f8bb03",
                "borderRadius": 16,
                "height": 10,
                "marginTop": 8,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#4b4b4e",
                    "borderColor": "#f4f4f6",
                    "borderRadius": 15,
                    "borderWidth": 5,
                    "elevation": 4,
                    "height": 24,
                    "left": -14,
                    "position": "absolute",
                    "top": -7,
                    "width": 24,
                  },
                  Object {
                    "left": "0%",
                  },
                ]
              }
            />
          </ViewManagerAdapter_ExpoLinearGradient>
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 16,
                "paddingLeft": 4,
                "paddingRight": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#9a9a9a",
                  "fontFamily": "Lato",
                  "fontSize": 13,
                }
              }
            >
              Light
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#9a9a9a",
                    "fontFamily": "Lato",
                    "fontSize": 13,
                  },
                  Object {
                    "marginLeft": 24,
                  },
                ]
              }
            >
              Moderate
            </Text>
            <Text
              style={
                Object {
                  "color": "#9a9a9a",
                  "fontFamily": "Lato",
                  "fontSize": 13,
                }
              }
            >
              Vigorous
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#2a2a2a",
                "borderRadius": 16,
                "flexGrow": 1,
                "margin": 8,
                "padding": 16,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "position": "absolute",
                "right": 16,
                "top": "50%",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#f4f4f6",
                    "fontSize": 32,
                  },
                  undefined,
                  Object {
                    "fontFamily": "material-community",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰎛
            </Text>
          </View>
          <Text
            style={
              Object {
                "color": "#f4f4f6",
                "fontFamily": "Lato",
                "fontSize": 15,
                "marginBottom": 4,
              }
            }
          >
            Note
          </Text>
          <Text
            style={
              Object {
                "color": "#f4f4f6",
                "fontFamily": "Lato",
                "fontSize": 15,
                "maxWidth": "90%",
              }
            }
          >
            Was really enjoying. Got into flow state.
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
